### PR TITLE
Fix failing test cases

### LINKED ISSUE #7: HowTo: Pass IGeometry as parameter to query
I implemented the ToGeometry static extension methods for Linq.
In the UnitTests where these functions play a role (the ones attributed Ignore("implement ToPolygon") I see that the IGeometry instance is being passed as parameter to the qurey and  not being converted to the db specific type. Therefor the tests fails.

Is there a way to instruct NHibernate to convert the parameter prior to passing it to the query?

### PR DESCRIPTION
**Pull request by peetw**
Fixes #7 and a bunch of other failing tests for PostGis20, MsSqlSpatial, and MsSql2008. However, it is based on the assumption that the `Tests.Nhibernate.Spatial.MsSqlSpatial` project is for testing the MsSql2012 dialect - is this correct?

Also, the NTS BooleanRelate test is still failing for MsSqlSpatial, even though it passes for MsSql2008 - I believe this is due to the [change in precision](https://blog.safe.com/2012/06/my-top-6-takeaways-from-sql-server-spatial-2012/) between 2008 and 2012. Before merging this branch, I would like to either ignore the test, or add a modified NTS test data - thoughts?

Once the above issue has been resolved, there should be no failing tests for the PostGis20, MsSqlSpatial, and MsSql2008 projects - let me know if that is not the case on the build server!

### CODE REVIEW

File: SharedAssemblyInfo.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 [assembly: AssemblyConfiguration("")]
 [assembly: AssemblyCompany("NHibernate.Spatial Community")]
 [assembly: AssemblyProduct("NHibernate.Spatial")]
-[assembly: AssemblyCopyright("Copyright � 2009-2016 Ricardo Stuven, Surya Pratap, Andreas Ravnestad, Pete Whitaker")]
+[assembly: AssemblyCopyright("Copyright � 2009-2017 Ricardo Stuven, Surya Pratap, Andreas Ravnestad, Peet Whitaker")]
```
Comment (andrerav) [inline SharedAssemblyInfo.cs]: Lol, sorry about that one :)

Comment (peetw) [inline SharedAssemblyInfo.cs]: No worries, most people think it's a typo :laughing:

**Comment by peetw** [comment]
I think I need to revert/modify some of these changes, based on [this comment](https://github.com/nhibernate/NHibernate.Spatial/issues/52#issuecomment-309717221). The `Tests.Nhibernate.Spatial.MsSqlSpatial` project should actually be testing against MS SQL 2005 databases using the `MsSqlLegacyDialect`, not MS SQL 2012... Thoughts?

**Comment by peetw** [comment]
I'm going to close this PR and open another one, but without the changes to the `MsSqlSpatial` test project, in anticipation of it being removed.

**Pull request closed without merging**